### PR TITLE
Feature 346 support for scheduled fields

### PR DESCRIPTION
**Pull request by Jdallos**

### CODE REVIEW

**Review by EVP1989 — COMMENTED** [review]
Added comments

File: src/components/form-editor/metaform-editor-right-drawer-visibility.tsx [hunk 1]
```diff
@@ -575,6 +607,135 @@ const MetaFormRightDrawerVisibility: FC<Props> = ({
...
+    const updatedForm = produce(pendingForm, draftForm => {
+      const startTime = draftForm.sections![metaformSectionIndex].fields![metaformFieldIndex].schedule?.startTime;
+
+      draftForm.sections![metaformSectionIndex].fields![metaformFieldIndex].schedule = {
+        startTime: startTime,
+        endTime: dateTime
+      };
+    });
+    setPendingForm(updatedForm);
+  };
```
[the remark this answers is not in the document]
Comment (Jdallos) [reply src/components/form-editor/metaform-editor-right-drawer-visibility.tsx]: I think this comment is outdated as this has already been done.

File: src/components/form-editor/metaform-editor-right-drawer-visibility.tsx [hunk 2]
```diff
@@ -575,6 +607,135 @@ const MetaFormRightDrawerVisibility: FC<Props> = ({
...
+                value={ scheduledVisibilityEnd ?? null }
+                onChange={ scheduledEndTimeHandler }
+                renderInput={ params =>
+                  <TextField { ...params }/>
+                }
+                views={["day", "hours"]}
+                label={ strings.draftEditorScreen.editor.schedule.endDate }
+                minDateTime={ scheduledVisibilityStart ?? undefined}
+              />
+            </LocalizationProvider>
```
[the remark this answers is not in the document]
Comment (Jdallos) [reply src/components/form-editor/metaform-editor-right-drawer-visibility.tsx]: Also already done

File: src/components/form-editor/metaform-editor-right-drawer-visibility.tsx [hunk 3]
```diff
@@ -53,8 +61,32 @@ const MetaFormRightDrawerVisibility: FC<Props> = ({
...
+   */
+  const loadScheduledVisibility = () => {
+    const field = MetaformUtils.getMetaformField(pendingForm, metaformSectionIndex, metaformFieldIndex);
+    if (!field) return null;
+
+    if (!field.schedule) {
+      setScheduledVisibility(undefined);
+      setScheduledVisibilityStart(null);
+      setScheduledVisibilityEnd(null);
+      return;
```
[the remark this answers is not in the document]
Comment (Jdallos) [reply src/components/form-editor/metaform-editor-right-drawer-visibility.tsx:75]: Done

File: src/utils/schedule-utils.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      if (endTime) {
+        const endDateTime = DateTime.fromJSDate(new Date(endTime));
+        return startDateTime <= currentTime && currentTime < endDateTime;
+      }
+    } else if (endTime) {
+      const endDateTime = DateTime.fromJSDate(new Date(endTime));
+      return currentTime < endDateTime;
+    }
+    return true;
+  };
```
[the remark this answers is not in the document]
Comment (Jdallos) [reply src/utils/schedule-utils.ts:42]: Refactored to use a switch

**Review by EVP1989 — APPROVED** [review]
Approved

**Pull request merged**